### PR TITLE
Feature bulk state change

### LINKED ISSUE #635: Bulk change registered state to findable
All DOIs for client TIB.UBBS registered between 26 June 2018 and 14 December 2018.

Front logo [Front conversations](https://app.frontapp.com/open/top_g6j7)

### PR DESCRIPTION
**Pull request by kjgarza**
to close https://github.com/datacite/datacite/issues/635

### CODE REVIEW

File: lib/cirneco/api.rb [hunk 1]
```diff
@@ -109,5 +109,28 @@ def get_media(doi, options={})
       end
       response
     end
+  
+    def update_rest_doi(doi, options={})
+      return OpenStruct.new(body: { "errors" => [{ "title" => "JWT or Username or password missing" }] }) unless options[:jwt].present? || (options[:username].present? && options[:password].present?)
+
+      api_url = options[:sandbox] ? 'https://api.test.datacite.org' : 'https://api.datacite.org'
```
Comment (richardhallett) [inline lib/cirneco/api.rb:116]: Can we move this to somewhere more central, you repeat it in every function.
Presumably there is some global config somewhere.

File: lib/cirneco/api.rb [hunk 2]
```diff
@@ -109,5 +109,28 @@ def get_media(doi, options={})
...
+      end
+    end
+
+    def get_rest_doi(doi, options={})
+      return OpenStruct.new(body: { "errors" => [{ "title" => "Username or password missing" }] }) unless options[:username].present? && options[:password].present?
+
+      api_url = options[:sandbox] ? 'https://api.test.datacite.org' : 'https://api.datacite.org'
+
+      url = "#{api_url}/dois/#{doi}"
+      reponse = Maremma.get(url, accept: 'application/vnd.api+json;charset=UTF-8', username: options[:username], password: options[:password], raw: true)
```
Comment (richardhallett) [inline lib/cirneco/api.rb]: *response

**Review by richardhallett — APPROVED** [review]
Overall looks fine to merge, just a couple minor re-factoring things that would be good to do before you merge.

File: lib/cirneco/api.rb (hunk 2 above)
Comment (kjgarza) [reply lib/cirneco/api.rb]: fixed

**Pull request merged**